tests/test_parser.py

```diff
@@ -107,3 +108,114 @@ def test() -> None:
             ],
         ),
     )
+
+    tree = sqltree("SELECT a::char FROM a", dialect=Dialect(Vendor.redshift))
+    assert tree == p.Select(
```

Comment:
These tests are good but really annoying to write. Feel free to add only formatter tests for future parser changes